src/p2p/p2p.cpp

```diff
@@ -90,7 +90,12 @@ void broadcast_message(const peer_outbound_message msg)
     //Broadcast while locking the peer_connections.
     std::lock_guard<std::mutex> lock(ctx.peer_connections_mutex);
     for (const auto &[k, session] : ctx.peer_connections)
+    {
+        if (!self_recieve && session->address == "0.0.0.0")
```

Comment:
`session->address == "0.0.0.0"`: Instead of this check, shall we introduce `is_self` boolean field into `socket_session`? We can set that field when the session is created.